internal/cloud/realm/auth_test.go

```diff
@@ -78,16 +79,58 @@ func TestRealmAuthRefresh(t *testing.T) {
 		assert.Equal(t, nil, err)
 
 		// invalidate the session's tokens
-		session.AccessToken = ""
 		session.RefreshToken = session.AccessToken
+		session.AccessToken = ""
 
 		profile := mock.NewProfileWithSession(t, session)
 
 		client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
 		_, err = client.AuthProfile()
 		assert.Equal(t, realm.ErrInvalidSession{}, err)
 	})
-	// TODO: REALMC-7719 add test for expired credentials and test for ensuring profile cleared on invalid session
+
+	t.Run("with an expired access token", func(t *testing.T) {
+		u.SkipUnlessExpiredAccessTokenPresent(t)
+		t.Run("should use the refresh token to generate a new access token", func(t *testing.T) {
+			client := realm.NewClient(u.RealmServerURL())
+			u.SkipUnlessRealmServerRunning(t)
```

Comment:
should be able to remove this, it's called at the top of `TestRealmAuthRefresh` (which I _think_ we're under here? hard to tell on github...)